networkit/cpp/centrality/test/CentralityGTest.cpp

```diff
@@ -584,6 +627,48 @@ TEST_F(CentralityGTest, testPageRankCentrality) {
     EXPECT_NEAR(0.0565, fabs(cen[7]), tol);
 }
 
+TEST_F(CentralityGTest, testNormalizedPageRankCentrality) {
+    /* Graph:
+     0    3   6
+      \  / \ /
+       2 -- 5
+      /  \ / \
+     1    4   7
+
+     Edges in the upper row have weight 3,
+     the edge in the middle row has weight 1.5,
+     edges in the lower row have weight 2.
+    */
+    count n = 8;
+    Graph G(n, true);
+
+    G.addEdge(0, 2, 3);
+    G.addEdge(1, 2, 2);
+    G.addEdge(2, 3, 3);
+    G.addEdge(2, 4, 2);
+    G.addEdge(2, 5, 1.5);
+    G.addEdge(3, 5, 3);
+    G.addEdge(4, 5, 2);
+    G.addEdge(5, 6, 3);
+    G.addEdge(5, 7, 2);
+
+    double damp = 0.85;
+    PageRank centrality(G, damp, 1e-8, true);
+    centrality.run();
+    std::vector<double> cen = centrality.scores();
+
+    // compare to Matlab results
+    const double tol = 1;
```

Comment:
It is because PageRank uses absolute error tolerance instead of relative. Since the test uses an undirected Graph the values for Normalized are incredibly high, as there are no dangling nodes in a connected undirected graph. So in terms of relative error tolerance, the error here is about equal to the one in the non-normalized PageRank test.

Reply:
That would be a much higher accuracy than the PageRank Test.

Because GTest's EXPECT_NEAR uses absolute tolerance. While regular PageRank goes from 0 to 1 normalized page rank goes from 1 to an upper limit that is solely graph dependant. In this case, the results are all over 40 million so if the result is 40 000 000.5 or 40 000 000.6 doesn't make that much difference. while the same test in Pagerank are 0.0565 or higher and use 1e-4(not 1e-6 or lower) is a much bigger fraction of the result than 1 is for the results of normalized Pagerank, so currently the tests for normalized PageRank are actually more precise than the ones for regular Page Rank

P.S. keep in mind that does not refer to the tolerance within the PageRank algorithm that still uses the default 1e-8 when calculating the PageRank since the normalization is the last step.

Reply:
My point is: if we have Matlab results that are precise up to `1e-4` why do we test with `eps = 1`? If our implementation is correct I would expect it to return (nearly) the same results as Matlab. Do we get different results compared to Matlab if we test with `eps = 1e-4`?